### ISSUE TITLE
ExtAuth should return a 401 for expired access_token

### ISSUE THREAD

**Issue by bcollard** [solo-io/gloo#6318]
### Gloo Edge Version

1.12.x (beta)

### Describe the bug

In the ExtAuth service, our JWT validation logic will return a 403 for any error instead of a 401. 

[Validator.go](https://github.com/solo-io/ext-auth-service/blob/26052235c2f91e9d0a0f77a340c636cd3aa2b324/pkg/config/oauth/token_validation/jwt/validator.go#L108)
[Auth_service.go](https://github.com/solo-io/ext-auth-service/blob/26052235c2f91e9d0a0f77a340c636cd3aa2b324/pkg/config/oauth/token_validation/auth_service.go#L60)

[RFC 6750](https://www.rfc-editor.org/rfc/rfc6750) states that the HTTP code should be 401 when the access token expires.

### Steps to reproduce the bug

```yaml
apiVersion: enterprise.gloo.solo.io/v1
kind: AuthConfig
metadata:
  name: hello-world
  namespace: apps
spec:
  configs:
  - oauth2:
      accessTokenValidation:
        jwt:
          remoteJwks:
            url: <URL_HERE>
```

### Expected Behavior

HTTP 401

**Comment by bcollard** [comment]
yes, it is urgent unless they manage to find a workaround. Customer is working in it now.

**Comment by jbohanon** [comment]
These changes were released as v0.19.19 in ext-auth-service, and @bewebi released v0.19.20 that same day for his CVE work. He has completed the dependency bump backports in gloo and is actively working on getting the open PRs for the dependency bump backports in solo-projects to merge. If they're still open tomorrow we're going to sync up again to assess further.
